pkg/trace/api/listener.go

```diff
@@ -95,6 +99,11 @@ func (sl *rateLimitedListener) Accept() (net.Conn, error) {
 
 // Close wraps the Close method of the underlying tcp listener
 func (sl *rateLimitedListener) Close() error {
+	if atomic.LoadUint32(&sl.closed) != 0 {
```

Comment:
If the objective is to prevent multiple calls shouldn't we do here 

```suggestion
	if !CompareAndSwapUint32(&sl.closeCalled, 0, 1) {
```

and remove the StoreUint32 in `Refresh`?

Otherwise, we could still have panics if `Closes` are called concurrently

Reply:
I like that. Changing now.